### PR TITLE
Adding filament style hidden option to columns

### PR DESCRIPTION
**Pull request by paulhennell**
Needed a way to hide columns for some users and instinctively reached for Filament style `->hidden()` options but it wasn't there so thought I'd add it.

Added tests to confirm it doesn't show when unintended, nor break any other functions. 

I had originally thought to use existing Filament\CanBeHidden trait, but there are multiple versions and all appear to add additional properties I am not confident would always work in this projects context. (See say `Filament\Schemas\Components\Concerns\CanBeHidden` which adds `hiddenOn` and many other options).

It seemed easier to simply duplicate only the few required options to a project CanBeHidden trait.

### CODE REVIEW

**Review by ManukMinasyan — CHANGES_REQUESTED** [review]
The `CanBeHidden` trait and API design look good -- clean implementation that follows Filament patterns. However, there's a consistency issue in `HasBoardColumns` that will cause bugs:

## Critical Issue

`getColumns()` now filters hidden columns, but **`getColumnIdentifiers()`, `getColumnLabels()`, and `getColumnColors()` still iterate over ALL columns** (including hidden ones). This creates array mismatches where these methods return data for columns that `getColumns()` excluded.

**Recommended fix:** Extract a `getVisibleColumns()` helper and use it in all getter methods:

```php
private function getVisibleColumns(): array
{
    return array_filter($this->columns, fn (Column $column) => $column->isVisible());
}

public function getColumns(): array
{
    return $this->getVisibleColumns();
}

public function getColumnIdentifiers(): array
{
    return array_map(fn (Column $column) => $column->getName(), $this->getVisibleColumns());
}

// Same for getColumnLabels(), getColumnColors()
```

## Missing Test Coverage

- No test for `->hidden(true)` with boolean literal (only Closure tested)
- No test for `getColumnIdentifiers()` / `getColumnLabels()` excluding hidden columns
- No test for edge case: all columns hidden

The trait itself is well-written -- just needs the getter consistency fix to be safe.

**Comment by ManukMinasyan** [comment]
Pushed three commits addressing all feedback:

- **3bd1ab5d** (@ManukMinasyan's critical issue, Copilot thread #1): added a `getVisibleColumns()` helper used by `getColumns()`, `getColumnIdentifiers()`, `getColumnLabels()`, and `getColumnColors()` so the maps stay in sync with the rendered columns. Wrapped the filter in `array_values()` so callers that JSON-encode the list never see a sparse, object-shaped payload.
- **fcae5098** (Copilot thread #2): documented the `CanBeHidden` evaluate() contract with a `@mixin \Filament\Support\Components\ViewComponent` hint and an explanatory docblock.
- **14c178d5** (@ManukMinasyan's missing coverage, Copilot thread #3): added a Column unit test covering `hidden(true)` and `visible(false)` literals, closure evaluation, and `hidden()`/`visible()` precedence; plus Livewire tests asserting the getters exclude hidden columns, that `getColumns()` returns a sequentially-indexed array, a re-hide-after-show regression, and the all-columns-hidden edge case.

Local checks (`vendor/bin/pint`, `vendor/bin/pest`) are green. PHPStan is clean on changed files — the 4 pre-existing `ternary.alwaysTrue` reports in `InteractsWithBoard.php` are present on `4.x` and unrelated to this PR.

File: src/Concerns/HasBoardColumns.php [hunk 1]
```diff
@@ -38,7 +38,7 @@ public function columns(array | Closure $columns): static
      */
     public function getColumns(): array
     {
-        return $this->columns;
+        return array_filter($this->columns, fn (Column $column) => $column->isVisible());
     }
```
[the remark this answers is not in the document]
Comment (ManukMinasyan) [reply src/Concerns/HasBoardColumns.php:45]: Addressed in 3bd1ab5d. The getter now wraps the filter in `array_values(...)` and delegates through a shared `getVisibleColumns()` helper, so the returned list is always sequentially indexed. The `fn (Column $column)` type hint is safe because `columns()` only accepts `Column` instances in practice, and the trait's `@var array<Column>` docblock enforces the contract for static analysers.

File: src/Concerns/CanBeHidden.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    public function isHidden(): bool
+    {
+        if ($this->evaluate($this->isHidden)) {
+            return true;
+        }
+
+        return ! $this->evaluate($this->isVisible);
+    }
```
[the remark this answers is not in the document]
Comment (ManukMinasyan) [reply src/Concerns/CanBeHidden.php:47]: Addressed in fcae5098. Added a `@mixin \Filament\Support\Components\ViewComponent` declaration and a docblock that explicitly states consuming classes must implement `evaluate()`.

File: tests/Feature/LivewireBoardTest.php [hunk 3]
```diff
@@ -13,7 +13,16 @@
...
+            ->assertDontSee('Hidden Column');
+    });
+
+    test('renders board with unhidden column', function () {
+        Livewire::test(TestBoard::class)
+            ->assertStatus(200)
+            ->assertDontSee('Hidden Column')
+            ->set('showAllColumns', true)
+            ->assertSee('Hidden Column');
     });
```
[the remark this answers is not in the document]
Comment (ManukMinasyan) [reply tests/Feature/LivewireBoardTest.php:27]: Addressed in 14c178d5. Added a new `tests/Unit/ColumnVisibilityTest.php` that exercises `hidden(true)`, `visible(false)`, closure evaluation, and the precedence between `hidden()` and `visible()`. Also added the suggested `re-hides column when visibility condition flips back` feature test to `LivewireBoardTest`.
